### PR TITLE
Add initial code

### PR DESCRIPTION
**Pull request by wrefgtzweve**

### CODE REVIEW

File: lua/autorun/server/sv_dfa.lua [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    end
+
+    if speed > punishSpeed then
+        damageVehicle( veh, driver, speed )
+    end
+end
+
+local function runCheck()
+    if not next( activeVehicles ) then return end
+
```
Comment (plally) [inline lua/autorun/server/sv_dfa.lua]: You dont need this unless it provides some significant performance increase

File: lua/autorun/server/sv_dfa.lua [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    local blackoutStart = warningStartOffset - punishSpeed
+    local blackoutScale = math.Clamp( speed + blackoutStart, 0, clampMagicNumber ) -- makes it get blacker faster
+    local blackoutAmount = blackoutScale * 2.1
+
+    if speed > blackoutStart then
+        driver:SetNWInt( "DFA_BlackingOut", blackoutAmount )
+    else
+        if driver:GetNWInt( "DFA_BlackingOut" ) ~= 0 then
+            driver:SetNWInt( "DFA_BlackingOut", 0 )
+        end
```
Comment (plally) [inline lua/autorun/server/sv_dfa.lua]: you can use elseif

**Review by plally — DISMISSED** [review]

File: lua/autorun/server/sv_dfa.lua (hunk 1 above)
Comment (StrawWagen) [reply lua/autorun/server/sv_dfa.lua]: Applied

File: lua/autorun/server/sv_dfa.lua (hunk 2 above)
Comment (StrawWagen) [reply lua/autorun/server/sv_dfa.lua]: Applied

File: lua/autorun/client/cl_dfa.lua [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    end
+
+    doBlackoutOverlayEffect( blackoutAlpha )
+
+    local finalAlphaClamped = math.Clamp( blackoutAlpha, 0, 250 ) -- players can always see a bit
+
+    surface_SetDrawColor( 0, 0, 0, finalAlphaClamped )
+    surface_DrawRect( 0, 0, ScrW(), ScrH() )
+
+end )
```
Comment (wrefgtzweve) [inline lua/autorun/client/cl_dfa.lua]: image

**Pull request merged**